Userland/DynamicLoader/main.cpp

```diff
@@ -15,16 +15,10 @@
 #include <unistd.h>
 
 char* __static_environ[] = { nullptr }; // We don't get the environment without some libc workarounds..
+char** environ = __static_environ;
 
-static void init_libc()
-{
-    environ = __static_environ;
-    __environ_is_malloced = false;
-    __stdio_is_initialized = false;
-    // Initialise the copy of libc included statically in Loader.so,
-    // initialisation of the dynamic libc.so is done by the DynamicLinker
-    __libc_init();
-}
+// FIXME: Kernel should give us a random value for __stack_chk_guard.
+uintptr_t __stack_chk_guard = 0xe0e6'066b'b7ea'c300;
```

Comment:
We already have 16 bytes of random bytes in the ELF Auxiliary Vector though?

https://github.com/SerenityOS/serenity/blob/cc92c3f5518d3d6effaa45d33ed661dcc5fb351f/Kernel/Syscalls/execve.cpp#L122-L127

Reply:
Oh wow, why have been ignoring them then? :caret:

Reply:
Not sure, they've been there for almost 4 years now :sweat_smile: 

https://github.com/SerenityOS/serenity/pull/2699/files#diff-6b86e3875f7596d11bf62500290c26f00f8aeb3a0c112bb26a720f497e92bc0aR1123